server/etcdserver/corrupt.go

```diff
@@ -184,15 +183,15 @@ func (s *EtcdServer) checkHashKV() error {
 		})
 	}
 
-	if h2 != h && rev2 == rev && crev == crev2 {
+	if h2 != h && h2.Revision == h.Revision && h.CompactRevision == h2.CompactRevision {
```

Comment:
```suggestion
	if h2.Hash != h.Hash && h2.Revision == h.Revision && h.CompactRevision == h2.CompactRevision {
```
I though I fixed that :(

Reply:
There is no difference. The suggested change has the same effect as the original implementation.

Reply:
ok, but that was not intentional :P